src/smirnoffio.py

```diff
@@ -58,49 +58,38 @@
-    for target in targets:
-        off_topology = None
-        ## remote targets are not initialized yet, we do a manual setup here
-        if isinstance(target, forcebalance.target.RemoteTarget):
-            if target.r_tgt_opts['type'].endswith('SMIRNOFF'):
-                target_path = os.path.join(target.root, target.tgtdir)
-                if target.r_tgt_opts['type'] == 'OPTGEOTARGET_SMIRNOFF':
-                    # parse optgeo_options_txt and get the names of the mol2 files
-                    optgeo_options_txt = os.path.join(target_path, target.r_tgt_opts['optgeo_options_txt'])
-                    sys_opts = forcebalance.opt_geo_target.OptGeoTarget.parse_optgeo_options(optgeo_options_txt)
-                    openff_mols = [OffMolecule.from_file(os.path.join(target_path,fnm), allow_undefined_stereo=True) \
-                                   for sysopt in sys_opts.values() for fnm in sysopt['mol2']]
-                else:
-                    openff_mols = [OffMolecule.from_file(os.path.join(target_path,fnm), allow_undefined_stereo=True) \
-                                   for fnm in target.r_tgt_opts.get('mol2', [])]
-                off_topology = OffTopology.from_molecules(openff_mols)
-        elif isinstance(target, forcebalance.opt_geo_target.OptGeoTarget):
-            if target.engine_.__name__ == 'SMIRNOFF':
-                target_path = os.path.join(target.root, target.tgtdir)
-                openff_mols = [OffMolecule.from_file(os.path.join(target_path,fnm), allow_undefined_stereo=True) \
-                    for sysopt in target.sys_opts.values() for fnm in sysopt['mol2']]
-                off_topology = OffTopology.from_molecules(openff_mols)
-        elif hasattr(target, 'engine') and isinstance(target.engine, SMIRNOFF) and hasattr(target.engine,'off_topology'):
-            off_topology = target.engine.off_topology
-        if off_topology is not None:
+    for tgt_option in tgt_opts:
+        target_path = os.path.join('targets', tgt_option['name'])
+        # aggregate mol2 file paths from all targets
+        mol2_paths = []
+        if tgt_option['type'] == 'OPTGEOTARGET_SMIRNOFF':
+            # parse optgeo_options_txt and get the names of the mol2 files
+            optgeo_options_txt = os.path.join(target_path, tgt_option['optgeo_options_txt'])
+            sys_opts = forcebalance.opt_geo_target.OptGeoTarget.parse_optgeo_options(optgeo_options_txt)
+            mol2_paths = [os.path.join(target_path,fnm) for sysopt in sys_opts.values() for fnm in sysopt['mol2']]
+        elif tgt_option['type'].endswith('_SMIRNOFF'):
+            mol2_paths = [os.path.join(target_path,fnm) for fnm in tgt_option['mol2']]
+        # analyze SMIRKs terms
+        for mol_fnm in mol2_paths:
+            # we work with one file at a time to avoid the topology sliently combine "same" molecules
+            openff_mols = [OffMolecule.from_file(mol_fnm)]
+            off_topology = OffTopology.from_molecules(openff_mols)
```

Comment:
As a small detail, I'd move defining the list `[openff_mols]` to inside of `OffTopology.from_molecule([openff_mol])`, to make it more consistent with how it's used in other consortium code. This is a style suggestion and you can feel free to ignore it.

Reply:
Sure it's updated. The `openff_mols` list was used to aggregate all related `mol2` files into a `off_topology`, and run `label_molecules(off_topology)`. I later realized that the `off_topology` automatically removes the duplicated molecules, which I don't want. Therefore I changed that to work with one `mol2` file at a time.

Reply:
Great. Thanks for the explanation -- that makes sense.